components/automate-ui/src/app/pages/policy/add-members/policy-add-members.component.ts

```diff
@@ -305,4 +332,74 @@ export class PolicyAddMembersComponent implements OnInit, OnDestroy {
     this.addOrRemoveQueuedMember(true, member);
     this.closeModal();
   }
+
+  public showInputs(inputName: string): boolean {
+    const typeValue = this.expressionForm.get('type').value;
+    const identityValue = this.expressionForm.get('identity').value;
+
+    switch (inputName) {
+      case 'identity':
+        if (typeValue === 'USER' || typeValue === 'TEAM') {
+            this.allIdentity = typeValue;
+            this.updateValidations('identity', true);
+            return true;
+          } else {
+            this.updateValidations('identity', false);
+            this.expressionForm.get('identity').reset();
+            return false;
+          }
+        break;
+      case 'name':
+        if ( typeValue === 'TOKEN' || (identityValue && identityValue !== '*') ) {
+          this.updateValidations('name', true);
+          return true;
+        } else {
+          this.updateValidations('name', false);
+          this.expressionForm.get('name').reset();
+          return false;
+        }
+        break;
+      default:
+        return false;
+    }
+  }
+
+  private updateValidations(inputName: string, active: boolean): void {
+    const updateInput = this.expressionForm.get(inputName);
+
+    active === true
+      ? updateInput.setValidators([Validators.required,
+        Validators.pattern(Regex.patterns.NON_BLANK)])
+      : updateInput.clearValidators();
+
+      updateInput.updateValueAndValidity();
+    }
+
+
+  private displayExpressionOutput(formValues: object): void {
+    const entries = Object.entries(formValues);
+    const output = [];
+
+    entries.forEach( ([key, value]) => {
+      if (key === 'name' && value != null && value.length > 0) {
+        return output.push(value);
+      } else if (value != null && value.length > 0) {
+        return output.push(value.toLowerCase());
+      }
+    });
+
+    this.expressionOutput = output.join(':');
+  }
+
+  private subscribeToExpressionFormChanges(): void {
+    // initialize stream
+    const expressionValueChanges = this.expressionForm.valueChanges;
+    // subscribe to form changes
+    this.formSubscription = expressionValueChanges.subscribe(formValues =>
+              this.displayExpressionOutput(formValues));
+  }
+
+  private unsubscribeFromExpressionFormChanges(): void {
+    this.formSubscription.unsubscribe();
+  }
```

Comment:
Where these are used are already tiny functions; would suggest to inline the code of these two functions without losing any clarity. (Also, don't think the comments on L395 and L397 are needed.)